src/api/entities/confidential/__tests__/ConfidentialTransaction/index.ts

```diff
@@ -398,6 +406,158 @@ describe('ConfidentialTransaction class', () => {
     });
   });
 
+  describe('method: affirm', () => {
+    it('should prepare the procedure and return the resulting transaction', async () => {
+      const expectedTransaction =
+        'someTransaction' as unknown as PolymeshTransaction<ConfidentialTransaction>;
+
+      const args = { legId, party: ConfidentialAffirmParty.Mediator } as const;
+
+      when(procedureMockUtils.getPrepareMock())
+        .calledWith(
+          {
+            args: { transaction, ...args },
+            transformer: undefined,
+          },
+          context,
+          {}
+        )
+        .mockResolvedValue(expectedTransaction);
+
+      const tx = await transaction.affirmLeg(args);
+
+      expect(tx).toBe(expectedTransaction);
+    });
+  });
+
+  describe('method: reject', () => {
+    it('should prepare the procedure and return the resulting transaction', async () => {
+      const expectedTransaction =
+        'someTransaction' as unknown as PolymeshTransaction<ConfidentialTransaction>;
+
+      when(procedureMockUtils.getPrepareMock())
+        .calledWith(
+          {
+            args: { transaction },
+            transformer: undefined,
+          },
+          context,
+          {}
+        )
+        .mockResolvedValue(expectedTransaction);
+
+      const tx = await transaction.reject();
+
+      expect(tx).toBe(expectedTransaction);
+    });
+  });
+
+  describe('leg state methods', () => {});
+  let mockLegState: dsMockUtils.MockCodec<PalletConfidentialAssetTransactionLegState>;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let mockLegReturn: any;
+
+  beforeEach(() => {
+    mockLegState = dsMockUtils.createMockConfidentialLegState({
+      assetState: dsMockUtils.createMockBTreeMap<
+        Bytes,
+        PalletConfidentialAssetTransactionLegState
+      >(),
+    });
+    mockLegReturn = dsMockUtils.createMockOption(mockLegState);
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (mockLegState.assetState as any).toJSON = (): Record<string, ConfidentialLegStateBalances> => ({
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      '0x01': {
+        senderInitBalance: '0x02',
+        senderAmount: '0x03',
+        receiverAmount: '0x04',
+      },
+    });
+
+    jest.spyOn(mockLegReturn, 'unwrap').mockReturnValue(mockLegState);
```

Comment:
Why is there a need to spy the unwrap? I think if you set the assetState.toJSON() before initializing the mockLegReturn it could work as expected